File: 100+_python_Projects/weatherCropAnalytics/Initial Analysis/FetchWeather.js
async function fetchTemperatureAndHumidityData(latitude, longitude) {
    const apiUrl = `https://api.open-meteo.com/v1/forecast?latitude=${latitude}&longitude=${longitude}&hourly=temperature_2m,relativehumidity_2m&forecast_days=1`;

    try {
        const response = await fetch(apiUrl);
        if (response.ok) {
            const data = await response.json();
            const hourlyData = data.hourly[0];
            if (hourlyData) {
                const temperature = hourlyData.temperature_2m[0];
                const humidity = hourlyData.relativehumidity_2m[0];

                return {
                    temperature: temperature,
                    humidity: humidity
                };
            } else {
                console.error("No temperature and humidity data found in the response.");
                return null;
            }
        } else {
            console.error(`Failed to fetch weather data. Status Code: ${response.status}`);
            return null;
        }
    } catch (error) {
        console.error("An error occurred:", error);
        return null;
    }
}

// Usage example with provided latitude and longitude
const latitude = 52.52; //  latitude
const longitude = 13.41; //  longitude

fetchTemperatureAndHumidityData(latitude, longitude)
    .then(weatherData => {
        if (weatherData) {
            console.log("Weather Data:");
            console.log(`Temperature: ${weatherData.temperature}°C`);
            console.log(`Humidity: ${weatherData.humidity}%`);
        }
    });

File: 100+_python_Projects/weatherCropAnalytics/Initial Analysis/fetchWeather.py
import requests

def fetchTemperatureAndHumidityData(latitude, longitude):
    url = f"https://api.open-meteo.com/v1/forecast?latitude={latitude}&longitude={longitude}&hourly=temperature_2m,relativehumidity_2m&forecast_days=1"
    
    try:
        response = requests.get(url)
        print(response)
        data = response.json()
        
        if response.status_code == 200:
            hourly_data = data.get("hourly", [])
            
            # only fetch data when hourly status has been fetched
            if hourly_data:
                time = hourly_data.get("time", [])
                temperature = hourly_data.get("temperature_2m", [])
                humidity = hourly_data.get("relativehumidity_2m", [])
                
                structured_data = {
                    "time": time,
                    "temperature": temperature,
                    "humidity": humidity
                }
                
                return structured_data
            else:
                print("No hourly data found in the response.")
                return None
        else:
            print(f"Failed to fetch weather data. Status Code: {response.status_code}")
            return None
    except Exception as e:
        print(f"An error occurred: {e}")
        return None

# latitude and longitude 
latitude = 52.52  
longitude = 13.41  


weather_data = fetchTemperatureAndHumidityData(latitude, longitude)
if weather_data:
    print("Weather Data:")
    print(f"Temperature: {weather_data['temperature']}°C")
    print(f"Humidity: {weather_data['humidity']}%")
